src/main/java/com/example/alaa/university/service/ApplicationUserService.java

```diff
@@ -0,0 +1,27 @@
+package com.example.alaa.university.service;
+
+import com.example.alaa.university.repository.ApplicationUserRepo;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ApplicationUserService {
+    private final ApplicationUserRepo appUserRepo;
+
+    public ApplicationUserService(ApplicationUserRepo appUserRepo) {
+        this.appUserRepo = appUserRepo;
+    }
+
+    public UserDetailsService userDetailsService() {
```

Comment:
Tha is very strage. You can make this calss implement UserDetailsService and use this calss a UserDetailsService bean. Other vise everytime userDetailsService is called it creates a new instance of userDetailsService.